src/NodeContainer/ParsedNodesByType.php

```diff
@@ -484,6 +491,21 @@ private function addClassConstant(ClassConst $classConst): void
     }
 
     private function addClassConstantFetch(ClassConstFetch $classConstFetch): void
+    {
+        $this->classConstantFetches[] = $classConstFetch;
+    }
+
+    private function processClassConstantFetches(): void
+    {
+        if ($this->classConstantFetches) {
+            foreach ($this->classConstantFetches as $rawClassConstantFetch) {
+                $this->processClassConstantFetch($rawClassConstantFetch);
+            }
+            $this->classConstantFetches = [];
```

Comment:
Why is the proprety resetted?

Reply:
So that this is only executed once on the first call to `findClassConstantFetches`. Basically lazy-loading the data into `$classConstantFetchByClassAndName`. Once this is done we don't need to do it again, that's why `$this->classConstantFetches` is reset to an empty value.

Reply:
Could you think of better to add class constant fetches to prevent these time-dependency?

Reply:
I think the only proper way to fix it would be to introduce another step between "parsing" and "refactoring". In that step all the work for resolving dependencies would be done, once parsing has completed and therefore all information about classes/methods/properties/constants has been collected. The current approach is doing this during `collect` and implictly relies on parsing order.

Reply:
I'm trying to say that we have all we need now.

Look at: https://github.com/rectorphp/rector/blob/master/packages/DeadCode/src/Analyzer/SetterOnlyMethodAnalyzer.php#L68

It gets all the setter only methods just once before any refactoring, but after the parsing.

Reply:
Ah, I see. So this is the more reliable and cleaner approach to process information after parsing. I think I can refactor it into following that implementation pattern. I'll have a look at it this evening. Thanks for the hint!